src/reactviews/pages/PublishProject/publishProject.tsx

```diff
@@ -35,33 +36,56 @@ function PublishProjectDialog() {
     const context = useContext(PublishProjectContext);
 
     // Select pieces of state needed for this component
-    const formComponents = usePublishDialogSelector((s) => s.formComponents, Object.is);
-    const formState = usePublishDialogSelector((s) => s.formState, Object.is);
-    const inProgress = usePublishDialogSelector((s) => s.inProgress, Object.is);
-
+    const formComponents = usePublishDialogSelector((s) => s.formComponents);
+    const formState = usePublishDialogSelector((s) => s.formState);
+    const inProgress = usePublishDialogSelector((s) => s.inProgress);
     // Check if component is properly initialized and ready for user interaction
     const isComponentReady = !!context && !!formComponents && !!formState;
 
-    // Let the form framework handle validation - check if any visible components have validation errors
+    // Check if any visible component has an explicit validation error.
     const hasValidationErrors =
         isComponentReady && formComponents
             ? Object.values(formComponents).some(
                   (component) =>
-                      !component.hidden && component.validation && !component.validation.isValid,
+                      !component.hidden &&
+                      component.validation !== undefined &&
+                      component.validation.isValid === false,
               )
             : false;
 
-    // Buttons should be disabled when:
-    // - Component is not ready (missing context, form components, or form state)
-    // - Operation is in progress
-    // - Form has validation errors
-    const readyToPublish = !isComponentReady || inProgress || hasValidationErrors;
+    // Identify missing required values for visible components (treat empty string / whitespace as missing)
+    const hasMissingRequiredValues =
```

Comment:
All this should be handled automatically in the controller.  If you're handling it here, then there's no point in using the form framework.